docs/DeveloperGuide.md

```diff
@@ -1,38 +1,324 @@
+    * [**`Command`**](#command): Contains the changes or updates to be made.
+    * [**`UI`**](#ui-component): The UI of the App.
+    * [**`Parser`**](#parser): Breaks down user input into parameters accepted by the app and creates a Command Object.
+    * [**`Validator`**](): Checks if the input provided by the User is Valid.
+    * [**`Storage`**](#storage): Reads data from data files and writes data to data files, also stores in app memory.
+
+The Sequence Diagram below showcases the general Logic and Flow of the program from Launch till Exit.
+
+![Sequence Diagram](diagrams/SequenceDiagram.png)
+
+### UI component
+
+![UIClassUML](diagrams/UIClassUML.png)
+
+How the UI class works:
+* Based on the architecture sequence diagram, `Manager` class calls readCommand method in `UI` class and returns the UserInput.
+* Afterwards, the `Manager` class calls readParameters method in `UI` class and returns another UserInput.
+* Finally, the `Manager` class calls print method in `UI` class which in turn calls `Status` enum and returns the constant.
+
+### Manager component
+
+![ManagerUML](diagrams/ManagerClass.png)
+<br>
+How the Manager class works:
+* When `Duke` class instantiates a `Manager` object and calls runLoop method, the program will execute a while loop.
+* In the while loop, there is a switch condition.
+* Based on the parameter provided to the switch, it will call the respective methods in UI, Parser & Command classes.
+* The while loop only halts when isTerminated boolean becomes true. Then, the programme exits.
+
+### Helper Classes
+
+#### `Command`
+
+![CommandUML](diagrams/CommandClassUML.png)
+
+How the Command class works:
+* Based on the architecture sequence diagram, the `Parser` class calls the `Validator` class to check if the inputs are valid.
+* If it is valid, the `Command` subclass executes its method.
+* The subclass is based on whether the user wants to access either the `Doctor`, `Patient`, `Medicine` or `Appointment` asset list class.
+* The input parameters determine which of the `Command` subclass is used.
+
+#### `Validator`
+
+![ValidatorClassUML](diagrams/ValidatorClassUML.png)
+
+The validator has a series of methods to ensure that the parameters entered are correct.  It throws a `HalpmiException` if the parameters
+entered are invalid.
+For example, validateAddPatient validates the parameter of `add patient` command, ensuring each parameter is in correct
+format. Please refer to the below sequence diagram for a clearer understanding.
+
+![ValidatorUML](diagrams/ValidatorUML.png)
+
+
+#### `Parser`
+
+![ParserClassUML](diagrams/ParserClassUML.png)
+
+The parser parses the description of the command. It first checks the number of parameters entered is correct and
+calls the validator class to validate the parameters, and then returns a command if the validation is successful.
+
+![ParserUML](diagrams/ParserUML.png)
+
+#### `Storage`
+
+![StorageClassUML](diagrams/StorageClassUML.png)
```

Comment:
Maybe there are too many inheritance arrowheads and association arrows overlapping one another? Consider changing the formatting so that the either the Storage or List class is at the bottom to remove the overlaps.